src/Microsoft.Extensions.Configuration/IncludedConfigurationProvider.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Configuration
+{
+    public class IncludedConfigurationProvider : ConfigurationProvider
+    {
+        public IncludedConfigurationProvider(IConfiguration source)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+            int pathStart = 0;
+            var section = source as IConfigurationSection;
+            if (section != null)
+            {
+                pathStart = section.Path.Length + 1;
+            }
+            foreach (var child in source.GetChildren())
+            {
+                AddSection(child, pathStart);
+            }
+        }
+
+        private void AddSection(IConfigurationSection section, int pathStart)
```

Comment:
Yes, and I don't think it would be hard. `GetChildrenKeys()` looks a bit hard but in the end I think it should be easy to implement with `GetChildren()`. The only really weird thing about the contract is that `GetChildrenKeys()` takes a delimiter argument. I suspect we should loose it. I suspect if there is any coversion between the standard configuration delimiter and whatever the provider's underlying representation uses, that is implementation detail of the provider.